File: problems/Most Common Word/mostCommonWord.cs
public class Solution {
    public string MostCommonWord(string paragraph, string[] banned) {
        var str = stripPunctuation(paragraph.Trim());
        var words = str.Split(' ');
        var store = new Dictionary<string, int>();
        var bannedWords = new HashSet<String>(banned);
        var counter = 0;
        var result = "";

        foreach (string item in words) {
            if (String.IsNullOrWhiteSpace(item)) {
                continue;
            }

            string word = item.ToLower();

            if (!bannedWords.Contains(word)) {
                if (store.ContainsKey(word)) {
                    ++store[word];
                } else {
                    store.Add(word, 1);
                }
            }
        }

        foreach (var word in store) {
            if (counter < word.Value) {
                counter = word.Value;
                result = word.Key;
            }
        }

        return result;
    }

    private string stripPunctuation(String str) {
        var sb = new StringBuilder();

        foreach (char letter in str) {
            if (char.IsPunctuation(letter)) {
                sb.Append(' ');
            } else {
                sb.Append(letter);
            }
        }

        return sb.ToString();
    }
}

File: problems/Most Common Word/mostCommonWord.java
class Solution {
    public String mostCommonWord(String paragraph, String[] banned) {
        Set<String> store = new HashSet<>(Arrays.asList(banned));
        HashMap<String, Integer> map = new HashMap<>();
        String[] words = paragraph.split("[\\p{Punct}\\s]+");
        int frequency = 0;
        String result = "";

        for (String word : words) {
            if (0 == word.trim().length()) {
                continue;
            }

            String lowCaseWord = word.toLowerCase();

            if (!store.contains(lowCaseWord)) {
                if (map.containsKey(lowCaseWord)) {
                    map.put(lowCaseWord, 1 + map.get(lowCaseWord));
                } else {
                    map.put(lowCaseWord, 1);
                }
            }
        }

        for (Map.Entry<String, Integer> word : map.entrySet()) {
            if (word.getValue() > frequency) {
                frequency = word.getValue();
                result = word.getKey();
            }
        }

        return result;
    }
}
